store/store.go

```diff
@@ -1178,6 +1198,43 @@ var download = func(name, downloadURL string, user *auth.UserState, s *Store, w
 	return err
 }
 
+// downloadDelta downloads the delta for the preferred format, returning the path.
+func (s *Store) downloadDelta(name string, downloadDir string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) (string, error) {
+
+	if len(downloadInfo.Deltas) != 1 {
+		return "", errors.New("Store returned more than one delta")
```

Comment:
Sorry - you did mention that earlier for logs, but managed to forget it when creating those errors. Done.